src/mlpack/core/util/arma_traits.hpp

```diff
@@ -163,6 +163,15 @@ struct GetColType
   typedef arma::Row<typename MatType::elem_type> type;
 };
 
+template<typename MatType>
+struct GetUIntColType
+{
+  typedef typename MatType::elem_type eT;
+  typedef typename std::conditional<
+    std::is_same<eT, arma::uword>::value, eT, arma::uword>::type elem;
```

Comment:
The `std::conditional` isn't doing anything here, why not just have `arma::Col<arma::uword>` as the result?

Reply:
Good advice, thanks